src/main/java/Duke.java

```diff
@@ -1,10 +1,193 @@
+        }
+        Task task = this.taskList.get(index);
+        task.mark();
+        return "Nice! I've marked this task as done:\n" + task.toString();
+    }
+
+    public String mark_as_undone(int index) throws DukeException {
+        if (taskList.size() <= index) {
+            throw new WrongIndexException();
+        }
+        Task task = this.taskList.get(index);
+        task.unmark();
+        return "OK, I've marked this task as not done yet:\n" + task.toString();
+    }
+
+    public String delete_msg(int index) throws WrongIndexException {
+        if (taskList.size() <= index) {
+            throw new WrongIndexException();
+        }
+        Task task = taskList.get(index);
+        taskList.remove(index);
+        return "Noted. I've removed this task:\n " + task.toString() +
+                "\nNow you have " + taskList.size() + " tasks in the list.";
+
+    }
+
+    public void loadDataFromDisk() {
+        storage.loadData(commandList);
+        for (String userInput: commandList) {
+            String[] expressions = userInput.split(" ");
+            String command = expressions[0];
+
+            try {
+                if (userInput.equals(parser.convertEnum(Command.LIST))) {
+                    continue;
+                } else if (command.equals(parser.convertEnum(Command.MARK))) {
+                    parser.checkEmpty(userInput, command);
+                    String[] words = userInput.split(" ");
+                    int index = Integer.parseInt(words[1]) - 1;
+                    String tmp = mark_as_done(index);
+                } else if (command.equals(parser.convertEnum(Command.UNMARK))) {
+                    parser.checkEmpty(userInput, command);
+                    String[] words = userInput.split(" ");
+                    int index = Integer.parseInt(words[1]) - 1;
+                    String tmp = mark_as_undone(index);
+                } else if (userInput.equals(parser.convertEnum(Command.BYE))) {
+                    continue;
+                } else if (command.equals(parser.convertEnum(Command.TODO))) {
+                    parser.checkEmpty(userInput, command);
+                    Todo todo_task = new Todo(userInput);
+                    addTask(todo_task);
+                    String tmp = msg_of_add(todo_task);
+                } else if (command.equals(parser.convertEnum(Command.DEADLINE))) {
+                    parser.checkEmpty(userInput, command);
+                    Deadline ddl_task = new Deadline(userInput);
+                    addTask(ddl_task);
+                    String tmp = msg_of_add(ddl_task);
+                } else if (command.equals(parser.convertEnum(Command.EVENT))) {
+                    parser.checkEmpty(userInput, command);
+                    Event event_task = new Event(userInput);
+                    addTask(event_task);
+                    String tmp = msg_of_add(event_task);
+                } else if (command.equals(parser.convertEnum(Command.DELETE))) {
```

Comment:
Maybe you would like to use a switch statement instead of a if-else statement? I personally uses a switch clause, which I feel that it makes it neater overall. 💭